### PR TITLE
Blog Route

### LINKED ISSUE #28: Band: Edit Blog Post
As a band I want to edit existing blog posts and create new ones.

Done when:

The band can see their existing blog posts
Clicking on a blog post displays text in an editable box
There's a save button that persists changes on the backend
There's a cancel button that doesn't make any changes
A create new button opens up an empty text box

### PR DESCRIPTION
**Pull request by tetron225**
Finished setting up a route for blogs would need to review

### CODE REVIEW

File: server/controllers/blog.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  const count = 10;
+  return Blog.findAll({
+    limit: count,
+  })
+    .then((result) => {
+      res.send(result);
+      next();
+    })
+    .catch((err) => {
+      res.send(500);
```
Comment (zkodjo) [inline server/controllers/blog.js:20]: Can you add logging for errors? Import the winston config file like this
```js
const logger = require('../../config/winston');
```

File: server/controllers/blog.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    where: {
+      id: bandID,
+    },
+  })
+    .then((result) => {
+      res.send(result);
+      next();
+    })
+    .catch((err) => {
+      res.send(500);
```
Comment (zkodjo) [inline server/controllers/blog.js:39]: Add logging per above

File: server/controllers/blog.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    where: {
+      id: bandID,
+    },
+  })
+    .then(() => {
+      res.status(200).send('Successfully Added');
+      next();
+    })
+    .catch((err) => {
+      res.send(400);
```
Comment (zkodjo) [inline server/controllers/blog.js:60]: Add logging

File: server/controllers/blog.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
+const { Op } = require('sequelize');
+const { Blog } = require('../models');
+
+//  getting all the blog post limited to first 10
+
+exports.getAllBlog = (req, res, next) => {
+  const count = 10;
```
Comment (kalebal) [inline server/controllers/blog.js]: I think the limit is a good idea, but we'll need a way to fetch pages as well if a band has more than 10 blog posts.

File: server/controllers/blog.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+exports.updateBlog = (req, res, next) => {
+  const bandID = req.params.id;
+  const updatePost = req.body.post;
+  return Blog.update({ post: updatePost }, {
+    where: {
+      id: bandID,
+    },
+  })
+    .then(() => {
+      res.status(200).send('Successfully Added');
```
Comment (zkodjo) [inline server/controllers/blog.js]: Can you change the response code to 201 since this is a put request?

File: server/controllers/blog.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+// add a new blog post from the specific user
+
+exports.addNewBlog = (req, res, next) => Blog.create({
+  name: req.body.name,
+  description: req.body.description,
+  post: req.body.post,
+  bandId: req.body.bandId,
+})
+  .then(() => {
+    res.status(200).send('Successfully Added a Blog post');
```
Comment (zkodjo) [inline server/controllers/blog.js]: Can you change the response code to 201 since this is a post request?

File: server/controllers/blog.js [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    next();
+  })
+  .catch((err) => {
+    res.send(400);
```
Comment (zkodjo) [inline server/controllers/blog.js:79]: Add error logging per above

**Review by kalebal — CHANGES_REQUESTED** [review]
This looks great! Only change is if there's a limit on the number of blog posts fetched we'll need a way to get different sets of 10, not just the first set. That way if there's more than 10 posts they can still be edited.

File: server/routes/blog.js [hunk 8]
```diff
@@ -0,0 +1,18 @@
+const express = require('express');
+
+const { getABlog, getAllBlog, updateBlog, addNewBlog } = requrie('../controllers/blog.js');
+
+const router = express.Router();
+//  getting all the blogs
+router.get('/blogs', getAllBlog);
+
+//  getting a single blog
+router.get('/blogs', getABlog);
```
Comment (zkodjo) [inline server/routes/blog.js]: Need to add a parameter here like :blogId or just :id or it won't work

File: server/routes/blog.js [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+
+//  posting a new blog
+router.post('/blogs', addNewBlog);
+
+//  updating an existing blog
+router.put('/blogs', updateBlog);
```
Comment (zkodjo) [inline server/routes/blog.js]: Need to add a parameter here so we know which blog post is updated. @kalebal does the client have access to the blog post id?

File: server/controllers/blog.js (hunk 4 above)
Comment (zkodjo) [inline server/controllers/blog.js]: Question: Should this be constant or do we want the client to be able to specify how many posts to get?

**Review by zkodjo — CHANGES_REQUESTED** [review]
Hey @tetron225 I have reviewed this and I left comments inline. Let's also discuss creating a seeder file to import some blog entries into the database

File: server/routes/blog.js (hunk 9 above)
Comment (kalebal) [reply server/routes/blog.js]: Currently just working with dummy data so no, but if the blog post id's are returned on get requests it can.

**Pull request closed without merging**